test/algorithms/ground_state_solvers/test_adapt_vqe.py

```diff
@@ -61,13 +66,67 @@ def setUp(self):
 
         self.qubit_converter = QubitConverter(ParityMapper())
 
+        self.inter_dist = 1.6
+
+        self.driver1 = PySCFDriver(
+            atom="Li .0 .0 .0; H .0 .0 " + str(self.inter_dist),
+            unit=UnitsType.ANGSTROM,
+            basis="sto3g",
+        )
+
+        self.transformer = ActiveSpaceTransformer(num_electrons=2, num_molecular_orbitals=3)
+
+        self.problem1 = ElectronicStructureProblem(self.driver1, [self.transformer])
+
     def test_default(self):
         """Default execution"""
         solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
         calc = AdaptVQE(self.qubit_converter, solver)
         res = calc.solve(self.problem)
         self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
 
+    def test_param_shift(self):
+        """test for param_shift method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="param_shift")
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_fin_diff(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="fin_diff", epsilon=1.0)
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_lin_comb(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="lin_comb", epsilon=1.0)
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_natural_gradients_invalid(self):
+        """test that an exception is thrown when an invalid gradient method is used"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = NaturalGradient(
+            grad_method="fin_diff", qfi_method="lin_comb_full", regularization="ridge"
+        )
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        with self.assertRaises(QiskitNatureError):
+            _ = calc.solve(self.problem)
```

Comment:
Just a comment for another future reviewer: we're aware that this is not optimal but currently we have not yet figured out a working scenario with `NaturalGradient`. This is planned to happen though when we migrate `AdaptVQE` to Terra as part of https://github.com/qiskit-advocate/qamp-spring-22/issues/3